internal/commands/app/diff.go

```diff
@@ -0,0 +1,138 @@
+package app
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/AlecAivazis/survey/v2"
+
+	"github.com/spf13/pflag"
+)
+
+type diffOutputs struct {
+	noDiffs bool
+}
+
+type diffInputs struct {
+	AppDirectory        string
+	IncludeDependencies bool
+	IncludeHosting      bool
+	cli.ProjectInputs
+}
+
+// CommandDiff is the `app diff` command
+type CommandDiff struct {
+	inputs      diffInputs
+	outputs     diffOutputs
+	atlasClient atlas.Client
+	realmClient realm.Client
+}
+
+// Flags is the command flags
+func (cmd *CommandDiff) Flags(fs *pflag.FlagSet) {
+	fs.StringVarP(&cmd.inputs.AppDirectory, flagDirectory, flagDirectoryShort, "", flagDirectoryUsage)
+	fs.BoolVarP(&cmd.inputs.IncludeDependencies, flagIncludeDependencies, flagIncludeDependenciesShort, false, flagIncludeDependenciesUsage)
+	fs.BoolVarP(&cmd.inputs.IncludeHosting, flagIncludeHosting, flagIncludeHostingShort, false, flagIncludeHostingUsage)
+
+	cmd.inputs.Flags(fs)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDiff) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandDiff) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandDiff) Handler(profile *cli.Profile, ui terminal.UI) error {
```

Comment:
just making a note for post-rebase: this signature will want to be updated to add the `clients cli.Clients` arg